articles/introduction-to-c#/index .md

```diff
@@ -0,0 +1,212 @@
+`Console.ReadLine()` is used to read the next characters in the input system.
+
+`Console. Read()` is used to read the next character in the input system.
+
+`Console.ReadKey()` is used to obtain the next key entered by the user. It is mostly used to hold the console for the user until he presses a key.
+
+### Comments in C#
+Comments start with two forward slashes. The compiler ignores comments when compiling the code.
+
+**Example**
+```c#
+// This is a C# program
+```
+
+This is a single-line comment. Comments are used by programmers to explain more about the line (what the line does). This concept is very important when a group of programmers is working together as it makes your code more readable to the others.
+
+For multiple-line comment, we use ` /*` and ends with `*/` :
+
+```C#
+/* C# is a programming language used to create Web applications and Mobile applications.
+
+It is an amazing language. */
+```
+
+Any text between `/*` and `*/` will be ignored by the compiler.
+
+**Note:**
+- Each line in C# ends with a semi-colon(;).
+- C# follows a top to bottom order when executing the program.
+- C# is case-sensitive as it treats Uppercase and Lowercase characters differently. e.g `program` and `Program` are different things.
+
+**Congratulations!** You just wrote your first C# program.
+
+Now, let's write a program that adds two numbers entered by the user.
+
+```C#
+//This is a C# program that adds two numbers
+
+using System;
+
+    namespace HelloWorld
+        {
+            class Program
+            {       
+                static void Main(string[] args){
+                    Console.WriteLine("Enter your first number: ");
+                    int num1= Convert.ToInt32(Console.ReadLine());
+                    Console.WriteLine("Enter your second number: ");
+                    int num2= Convert.ToInt32(Console.ReadLine());
+                    Console.WriteLine("Your Sum is "num1 + num2);//prints the sum of the two numbers 
+                    Console.ReadLine();
+               }
+         }
+    }
+```
+
+
+Here's how the output of the program will look like:
+
+```
+Enter your first number: 2
+
+Enter your second number: 5
+
+Your Sum is 7
+```
+
+Let us look at the new lines in our program:
+
+```c#
+int num1= Convert.ToInt32(Console.ReadLine());
+```
+
+`int num1` means that we are creating a memory location to hold an integer (int) and naming that memory location for future reference. In our case, we have named the memory location as **num1** that will hold an integer.
+
+Just like integers, we have a data type called `double` which is used to store fractional numbers:
+
+```C#
+double num1;
+```
+
+When we prompt the user to enter any details, C# takes the input as strings therefore, we have to convert the strings to numbers using the `Convert.ToInt32()` method above. 
+
+```C#
+Console.ReadLine()
+```
+
+It takes data from the user and allocates it to the memory location we had created earlier. 
```

Comment:
Console.ReadLine() alone doesn't take the data from the user and allocate it to the memory. Console.ReadLine() is just a function that returns the value from the Input system. What we do with the return value is up to the programmer.

`int num1= Convert.ToInt32(Console.ReadLine());` 

This entire statement is responsible for that action. In this case, we are using the return value from the `Console.ReadLine()` function and assigning it to `num1` using the `=` operator.